### PR TITLE
Set host MQTT environment

### LINKED ISSUE #576: Avoid use MQTT Address/Port hard code
**Is your feature request related to a problem? Please describe.**
Currently, in Flyvemdm we have the configuration of Mosquitto hard code.
**Describe the solution you'd like**
Set environment variables for the host and port.With this change we can change dinamicaly that parameters.

**Additional context**
This change is necesary for the configuration of the containers docker.
This is a reference:

https://github.com/flyve-mdm/docker-image/pull/31

### PR DESCRIPTION
**Pull request by ingluife**
Hi, @flyve-mdm/backend-development.

### Changes description

I changed the configuration of mosquitto service for the test and this plugin. With this change we can edit the parameters like host and port mqqt dinamically

### Estimated time

|Assignee|:tomato:|
|:---|:---:|
|@ingluife |4|
|@btry |2|

### References

Closes #576

### CODE REVIEW

**Comment by btry** [comment]
I know many projects do this, but I'm not in favor of this way to solve your problem. It is better to add arguments in the cliinstall.php script to read parameters from command line.

https://github.com/flyve-mdm/glpi-plugin/blob/develop/tools/cli_install.php

**Comment by ingluife** [comment]
Hi @btry @ajsb85. 

This is ready for merge.

File: tests/suite-unit/PluginFlyvemdmEntityConfig.php [hunk 1]
```diff
@@ -116,7 +116,7 @@ public function providerPrepareInputForUpdate() {
             [],
             [
                'You are not allowed to change the device limit',
-               'You are not allowed to download URL of the MDM agent',
+               'You are not allowed to change the download URL of the MDM agent',
```
Comment (btry) [inline tests/suite-unit/PluginFlyvemdmEntityConfig.php]: This should go in the PR I did today.

Comment (ingluife) [reply tests/suite-unit/PluginFlyvemdmEntityConfig.php]: That PR is merged?

**Comment by ingluife** [comment]
Hi, @btry.

> Please specify that addresses are either an IP address, either a hostname, in the docopt description.

This is ready.

File: tests/suite-unit/PluginFlyvemdmEntityConfig.php (hunk 1 above)
Comment (btry) [reply tests/suite-unit/PluginFlyvemdmEntityConfig.php]: I guess you fixed a unit test. If so, i t seems my PR did not passed tests but was merged anyway. I'll check this morning

**Comment by btry** [comment]
Added code refactor to validate the arguments. I tried to validate the IP / hostname with filter_var but I found unexpected results with the FILTER_DOMAIN filter. It seems it allows IPv4 and IPv6, especially a invalid IPv4 like '256.257.258.259', making tests fail. Removed the feature for now.

Also fixed an upgrade issue.

File: install/upgrade/update_to_dev.php [hunk 2]
```diff
@@ -178,7 +178,7 @@ function plugin_flyvemdm_update_to_dev(Migration $migration) {
       $migration->changeField($table, 'plugin_flyvemdm_fleets_id', 'items_id_applied', 'integer');
       $migration->dropKey($table, 'plugin_flyvemdm_fleets_id');
       $migration->addField($table, 'itemtype_applied', 'string', ['after' => 'id']);
-      $migration->addKey($table, 'FK_applied', ['itemtype_applied', 'items_id_applied']);
+      $migration->addKey($table, ['itemtype_applied', 'items_id_applied'], 'FK_applied');
```
Comment (btry) [inline install/upgrade/update_to_dev.php]: this call ws invalid: inverted arguments making upgrade fail.

File: tests/suite-unit/PluginFlyvemdmEntityConfig.php (hunk 1 above)
Comment (btry) [reply tests/suite-unit/PluginFlyvemdmEntityConfig.php]: My PR was merged despite unit tests failed. This line should fix the failed test.

File: tools/cli_install.php [hunk 3]
```diff
@@ -91,6 +95,29 @@
...
+   $pluginConfig = [
+      'mqtt_broker_port' => $args['--mqtt-port']
+   ];
+}
+if (isset($args['--mqtt-port-tls']) && $args['--mqtt-port-tls'] !== false) {
+   $pluginConfig = [
+      'mqtt_broker_tls_port' => $args['--mqtt-port-tls']
+   ];
+}
+
```
Comment (btry) [inline tools/cli_install.php]: This hunk prepares the configuration for the plugin in a single array

File: tools/cli_install.php [hunk 4]
```diff
@@ -168,6 +195,7 @@
       'download_url' => PLUGIN_FLYVEMDM_AGENT_BETA_DOWNLOAD_URL
    ]);
 }
+Config::setConfigurationValues('flyvemdm', $pluginConfig);
```
Comment (btry) [inline tools/cli_install.php]: Now we can apply the configuration we prepared earnier

**Pull request merged**